application/src/main/java/com/ericsson/bss/cassandra/ecchronos/application/DefaultNativeConnectionProvider.java

```diff
@@ -53,14 +51,45 @@ public DefaultNativeConnectionProvider(Config config, Supplier<Security.CqlSecur
             sslOptions = new ReloadingCertificateHandler(() -> cqlSecuritySupplier.get().getTls());
         }
 
-        myLocalNativeConnectionProvider = LocalNativeConnectionProvider.builder()
+        LocalNativeConnectionProvider.Builder nativeConnectionBuilder = LocalNativeConnectionProvider.builder()
                 .withLocalhost(host)
                 .withPort(port)
                 .withAuthProvider(authProvider)
-                .withSslOptions(sslOptions)
+                .withSslOptions(sslOptions);
+        establishTemporaryConnection(LocalNativeConnectionProvider.Builder.fromBuilder(nativeConnectionBuilder).build(),
+                host, port, nativeConfig.getTimeout());
```

Comment:
LocalNativeConnectionProvider.Builder.fromBuilder(...) seems a bit "into the internals".

Isn't it good enough to pass in the builder here and run "builder.build()" in the while loop?

That way we could also set myLocalNativeConnectionProvider to the result of the method immediately. The current implementation would leak a cluster/session object when finished.